src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.bookstore.repository.impl;
+
+import jakarta.persistence.criteria.CriteriaQuery;
+import java.util.List;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
```

Comment:
don't use try with resources and transaction, session will be closed before catch block and rollback won't happened if exception will thrown